src/screens/AddressDetail.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React from 'react';
+import AddressSummary from '../components/AddressSummary';
+import AddressHistory from '../components/AddressHistory';
+import { transaction, walletApi } from '@hathor/wallet-lib';
+
+
+class AddressDetail extends React.Component {
+  /*
+   * address {String} searched address (from url)
+   * errorMessage {String} message to be shown in case of an error
+   */
+  state = {
+    address: null,
+    errorMessage: '',
+  }
+
+  componentDidMount() {
+    // Expects address on URL
+    this.updateAddress(this.props.match.params.address);
+  }
+
+  componentDidUpdate(prevProps) {
+    if (prevProps.match.params.address !== this.props.match.params.address) {
```

Comment:
Shouldn't we create a new `AddressDetail` for every search? It would allow us to skip this.

Reply:
The problem is that the URL does not change, only the parameter, so it does not call the `componentDidMount` again. I can only catch the URL change with `componentDidUpdate`. Do you know a better way to do it?

Reply:
When the user searches for a new address, we can change the instance of the object of `AddressDetail`. So, react won't match with the current screen and will render the new one. I've done it in other projects. Maybe we should talk so I can understand if it really applies here, but I think it does.

Reply:
I still don't get it how I should do it without this check. When I am already in the search page and I just search for a new address, the method `this.props.history.push` won't trigger a new `componentDidMount`. This check is just to handle this situation

Reply:
I got it. The point is that the react is matching you `AddressDetail` object, instead of creating a new one. You can force react to create a new one, which is safer because you ensure it will get nothing from previous instance. I think that one way to do it is to put the search address in the key of the `AddressDetail`.

Reply:
`AddressDetail` is created in the `NavigationRoute`. I don't think it's common in react to have a key there. I would rather follow what they have in the `react-router` repository (https://github.com/ReactTraining/react-router/blob/c865bc6b331eabd853641dcc7e0224a7dce76f3b/docs/guides/ComponentLifecycle.md#fetching-data) where they capture url parameter changes exactly as I do in the explorer.

What do you think?

Reply:
Ok. I agree that you're following good practices.